homework1/task04/sum_of_four.py

```diff
@@ -0,0 +1,18 @@
+"""
+Classic task, a kind of walnut for you
+Given four lists A, B, C, D of integer values,
+    compute how many tuples (i, j, k, l) there are such that A[i] + B[j] + C[k] + D[l] is zero.
+We guarantee, that all A, B, C, D have same length of N where 0 ≤ N ≤ 1000.
+"""
+from typing import List
+
+
+def check_sum_of_four(a: List[int], b: List[int], c: List[int], d: List[int]) -> int:

```

Comment:
This is brute-force approach that will work in limited amount of cases.
This is an algorithmic task.
I will give you a hint: you can calculate how many possible sums give 2 first lists and second 2 lists.
If for example 3 possible pairs from first 2 lists give value of -2 and 4 different pairs in last 2 lists give +2, you have 12 different combinations that give value 0 in total sum.
this reduces looping to nested level of 2 max.

Reply:
another solution is to use itertools module - there's a function to iterate through all possible combinations of lists.
please write BOTH solutions, using itertools and utilizing algorithm from my previous comment